sdk/ai/ai-projects/src/tracing.ts

```
@@ -0,0 +1,278 @@
+    SUBMIT_TOOL_OUTPUTS = "submit_tool_outputs",
+    CREATE_THREAD_RUN = "create_thread_run",
+}
+
+export interface TracingAttributeOptions {
+    operationName?: string;
+    name?: string | null;
+    description?: string | null;
+    serverAddress?: string | null;
+    threadId?: string | null;
+    agentId?: string | null;
+    instructions?: string | null;
+    additional_instructions?: string | null;
+    runId?: string | null;
+    runStatus?: string | null;
+    responseModel?: string | null;
+    model?: string | null;
+    temperature?: number | null;
+    topP?: number | null;
+    maxPromptTokens?: number | null;
+    maxCompletionTokens?: number | null;
+    responseFormat?: string | null;
+    genAiSystem?: string | null;
+    messageId?: string | null;
+    messageStatus?: string | null;
+    eventContent?: string | null;
+    usagePromptTokens?: number | null;
+    usageCompletionTokens?: number | null;
+}
+
+export class TracingUtility {
+    private static tracingClient = createTracingClient({
+        namespace: "Microsoft.CognitiveServices",
+        packageName: "@azure/ai-projects",
+        packageVersion: "1.0.0-beta.1",
+
+    });
+
+    static async withSpan<Options extends { tracingOptions?: OperationTracingOptions },
+        Request extends (updatedOptions: Options) => ReturnType<Request>>(name: string, options: Options, request: Request,
+            startTrace?: (span: Omit<TracingSpan, "end">, updatedOptions: Options,) => void,
+            endTrace?: (span: Omit<TracingSpan, "end">, updatedOptions: Options, result: ReturnType<Request>) => void,
+        ): Promise<Resolved<ReturnType<Request>>> {
+        return TracingUtility.tracingClient.withSpan(name, options, async (updatedOptions: Options, span: Omit<TracingSpan, "end">) => {
+            if (startTrace) {
+                try {
+                    startTrace(span, updatedOptions);
+                }
+                catch { /* empty */ }
```

Same here, consider logging the error instead of silently dropping it